### PR TITLE
Endale.add new blog state

### PR DESCRIPTION
**Pull request by Endalebob**
The following tasks have been completed in this pull request:

Created and set up JSON server to mock API requests
Implemented API calls using RTK query to fetch data from the server
Configured Redux store and reducers for state management
Connected components to Redux store using React-Redux
Implemented actions and reducers to manage API data in the store
With this implementation, the application now has a mock API for testing and development purposes, and the state management has been refactored to use React-Redux.

Please review and merge this pull request.

Note: There are no changes to the UI in this pull request, so I did not include any screenshots.

### CODE REVIEW

File: Web/starter_project_vanguard/pages/api/blogs.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Blog } from '@/types/blog/blog'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import blogsJsonData from '../../data/blogs.json'
```
Comment (TsehaynehGetaneh) [inline Web/starter_project_vanguard/pages/api/blogs.ts:3]: use @ symbol to import files just for consistency

File: Web/starter_project_vanguard/store/features/blog/add-new-blog-api.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          url: '/blogs',
+          method: 'POST',
+          body: body,
+        }),
+      }),
+    }
+  },
+})
+
+export const { useAddBlogMutation } = addNewBlogApi
```
Comment (TsehaynehGetaneh) [inline Web/starter_project_vanguard/store/features/blog/add-new-blog-api.ts:24]: I am not sure but I don't think you need the features folder.

File: Web/starter_project_vanguard/pages/blog/add-new-blog.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      descriptionTitle:
+        'Network traffic passes through the front-end can seem hard at first glance. And you may not be familiar with advanced algorithms, but there are simple steps you can follow to see outstanding results in a short period of time.',
+      likes: 1722,
+      time: currentDate.getTime(),
+      date: 'Dec 13,2014',
+    })
+  }
+  useEffect(() => {
+    if (isAddBlogSuccess) {
+      // router.push("/home")
```
Comment (Rediet-Ferew) [inline Web/starter_project_vanguard/pages/blog/add-new-blog.tsx:98]: it is better to remove commented codes

File: Web/starter_project_vanguard/pages/blog/add-new-blog.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+  return (
+    <div className="flex flex-col lg:flex-row pl-20 mb-36 md:mb-44 lg:mb-72 pr-6 pt-6 min-h-screen">
+      <div className="w-full lg:w-3/4 pr-10 pb-10 mr-10">
+        <div>
+          <form className="mb-10" onSubmit={handleSubmit}>
+            <input
+              type="search"
+              id="search"
+              className="block w-full px-4 text-2xl md:text-3xl border-l-2 outline-none border-blue-700 font-montserrat"
```
Comment (Rediet-Ferew) [inline Web/starter_project_vanguard/pages/blog/add-new-blog.tsx:118]: use predefined colors in tailwind.config file

File: Web/starter_project_vanguard/pages/success-story/index.tsx [hunk 5]
```diff
@@ -1,10 +1,51 @@
 import Partners from '@/components/success-story/Partners'
 import SuccessDescription from '@/components/success-story/SuccessDescription'
 import SuccessImageCard from '@/components/success-story/SuccessImageCard'
-import { SuccessStory } from '@/types/success-story'
-import successStoryjsonData from '../../data/success-story.json'
+import { useFetchSuccessStoryQuery } from '@/store/features/success-story/success-story-api'
+import Link from 'next/link'
 export default function SucessStory() {
-  const successStories: SuccessStory[] = successStoryjsonData
+  const { data = [], isFetching } = useFetchSuccessStoryQuery()
```
Comment (Rediet-Ferew) [inline Web/starter_project_vanguard/pages/success-story/index.tsx:7]: rename data into descriptive name

File: Web/starter_project_vanguard/store/features/blog/add-new-blog-api.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { Blog } from '@/types/blog/blog'
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
+
+export const addNewBlogApi = createApi({
+  reducerPath: 'blog/add',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:3000/api' }),
```
Comment (Rediet-Ferew) [inline Web/starter_project_vanguard/store/features/blog/add-new-blog-api.ts:6]: use API_BASE_URL to name the api and use it as a baseurl

**Comment by Rediet-Ferew** [comment]
Looks good overall. Change the folder structure according to the guideline

**Review by Adona12 — APPROVED** [review]
Reviewed

**Pull request merged**